lib/Ix/Processor/JMAP.pm

```diff
@@ -80,22 +82,59 @@ has _dbic_handlers => (
   }
 );
 
-sub process_request ($self, $ctx, $calls) {
-  my @results;
+sub _sanity_check_calls ($self, $calls, $arg) {
+  # We should, in the future, add a bunch of error checking up front and reject
+  # badly-formed requests.  For now, this is a placeholder, except for its
+  # client id fixups. -- rjbs, 2018-01-05
+
+  my %saw_cid;
+
+  # Won't happen.  Won't happen.  Won't happen... -- rjbs, 2018-01-05
+  Carp::confess("too many method calls") if @$calls > 5_000;
+
+  for my $call (@$calls) {
```

Comment:
I think there's some danger here of this causing problems if someone interleaves calls that do and do not have cids.

Even if the answer is "Don't do that", perhaps we should add protection anyway?

```
  [ setFoo {}, ]  # We randomly pick x1342
  [ setFoo {}, ]  # We randomly pick x1
  [ setFoo {}, x1 ] # User provided x1 - collision!
```